copy/mkdir_windows.go

```diff
@@ -19,3 +30,64 @@ func fixRootDirectory(p string) string {
 func Utimes(p string, tm *time.Time) error {
 	return nil
 }
+
+func Chown(p string, old *User, fn Chowner) error {
+	if fn == nil {
+		return nil
+	}
+	user, err := fn(old)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	userSIDstring := user.SID
+	if userSIDstring == "" {
+		userSIDstring = containerAdministratorSidString
+
+	}
+	// Copy file ownership and ACL
+	// We need SeRestorePrivilege and SeTakeOwnershipPrivilege in order
+	// to restore security info on a file, especially if we're trying to
+	// apply security info which includes SIDs not necessarily present on
+	// the host.
+	privileges := []string{winio.SeRestorePrivilege, seTakeOwnershipPrivilege}
+	if err := winio.EnableProcessPrivileges(privileges); err != nil {
+		return err
+	}
+	defer winio.DisableProcessPrivileges(privileges)
+
+	sidPtr, err := syscall.UTF16PtrFromString(userSIDstring)
+	if err != nil {
+		return fmt.Errorf("converting to utf16 ptr: %w", err)
+	}
+	var userSID *windows.SID
+	if err := windows.ConvertStringSidToSid(sidPtr, &userSID); err != nil {
+		return fmt.Errorf("converting to windows SID: %w", err)
```

Comment:
Nit: use `errors.Errorf` so the timstamp does not get lost.

Reply:
@gabriel-samfira Could you fix this small issue and rebase. I think we can merge this then.